packages/engine-server/src/DendronEngineV3.ts

```diff
@@ -341,8 +343,46 @@ export class DendronEngineV3 implements DEngine {
     throw Error("querySchema not implemented");
   }
 
-  async queryNotes(): ReturnType<DEngineClient["queryNotes"]> {
-    throw Error("queryNotes not implemented");
+  /**
+   * See {@link DEngine.queryNotes}
+   */
+  async queryNotes(
+    opts: QueryNotesOpts
+  ): ReturnType<DEngineClient["queryNotes"]> {
+    const ctx = "Engine:queryNotes";
+    const { qs, vault, onlyDirectChildren, originalQS } = opts;
+
+    // Need to ignore this because the engine stringifies this property, so the types are incorrect.
+    // @ts-ignore
+    if (vault?.selfContained === "true" || vault?.selfContained === "false")
+      vault.selfContained = vault.selfContained === "true";
+
+    const noteIds = this.fuseEngine
+      .queryNote({
+        qs,
+        onlyDirectChildren,
+        originalQS,
+      })
+      .map((ent) => ent.id);
+
+    if (noteIds.length === 0) {
+      return { error: null, data: [] };
+    }
+
+    // TODO: Support createifNew? Doesn't seem to be actively used
```

Comment:
should not be used here, we should refactor so that lookup method will call `engine.create` if `queryNotes` returns empty